src/const_permutation.rs

```diff
@@ -0,0 +1,440 @@
+use core::fmt::Debug;
+
+use std::ops;
+
+#[cfg(feature = "random")]
+use rand::Rng;
+
+use crate::util::is_permutation;
+#[cfg(feature = "random")]
+use crate::Permutations;
+use crate::TryFromError;
+
+/// A permutation with a constant length fixed at compile time.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ConstPermutation<const N: usize>([usize; N]);
+impl<const N: usize> ConstPermutation<N> {
+    /// Returns the identity permutation.
+    pub fn identity() -> Self {
+        ConstPermutation((0..N).collect::<Vec<_>>().try_into().unwrap())
```

Comment:
It would be great if the const versions of these methods could avoid heap allocations and just allocate the array. E.g., here we could do something like

```suggestion
        let mut storage = [0; N];
        for i in 0..N {
            storage[i] = i;
        }
        Self(storage)
```